core/chain_makers.go

```diff
@@ -325,6 +325,11 @@ func (b *BlockGen) collectRequests(readonly bool) (requests [][]byte) {
 		// EIP-7251
 		ProcessConsolidationQueue(&requests, evm)
 	}
+
+	if b.cm.config.IsIsthmus(b.header.Time) {
+		requests = [][]byte{}
+	}
```

Comment:
If we move this above before `if b.cm.config.IsPrague(b.header.Number, b.header.Time) {` so it reads
```suggestion
	if b.cm.config.IsIsthmus(b.header.Time) {
		requests = [][]byte{}
	} else if b.cm.config.IsPrague(b.header.Number, b.header.Time) {
	// ...
```
then we can omit all the execution inside of the Prague-branch. Does that work?

Reply:
We could do that, but I think it makes sense to still run the EVM code to minimize functional diff between upstream geth and optimism. As implemented, requests will still be popped from the queue, but they'll always just ignored.

Reply:
I guess requests would always be empty anyways for an OP-Stack chain. So doesn't make much of a difference 👍🏻

Reply:
I agree with @meyer9. There's a semantic difference between not running that code and running it, and this actually affects execution witness generation too (given that accounts are touched in the above branch.) Base has worked _really_ hard on getting those witnesses to match up between clients, and it'd be a shame to cause more trouble than there already is there.

I think it's more simple for us to continue executing the upstream STF code and just ignore it with decoupled control flow.